DOCKER-network/dockernet

```diff
@@ -0,0 +1,275 @@
+    return build_yml_file
+
+
+def init(arguments):
+    print 'Initializing docker compose files...'
+
+    # If default flag is true then use default docker compose files
+    if arguments['default']:
+        shutil.copy('default/docker-compose-build.yml', '.')
+        shutil.copy('default/docker-compose.yml', '.')
+        return
+
+    # Remove unnecessary keys
+    arguments.pop('subparsers', None)
+    arguments.pop('default', None)
+
+    node_yml_file = generate_node_config(arguments)
+    with open(r'docker-compose.yml', 'w') as out_file:
+        yaml.dump(node_yml_file, out_file)
+
+    build_yml_file = generate_build_config(arguments)
+    with open(r'docker-compose-build.yml', 'w') as out_file:
+        yaml.dump(build_yml_file, out_file)
+
+    print 'done'
+
+
+def start(arguments):
+    print 'Starting up docker network...'
+    # Setup environment
+    docker_path = arguments['docker_path']
+    docker_bin = arguments['docker_bin']
+
+    if not os.path.exists(docker_path):
+        os.mkdir(docker_path)
+
+    if not os.path.exists(docker_bin):
+        os.mkdir(docker_bin)
+
+    if not os.path.exists('docker-compose-build.yml') or not os.path.exists('docker-compose.yml'):
+        print 'error: need to run > python dockernet init'
+        return
+
+    # Start build container and initialize build
+    os.system('docker-compose -f docker-compose-build.yml up -d')
+
+    # Wait for build to complete
+    print 'building binaries...'
+    while len(glob.glob(os.path.join(docker_path, '*-Node'))) == 0:
+        time.sleep(10)
+    print 'done'
+
+    # Wait for builder to finish
+    if arguments['deploy_smart_contracts']:
+        time.sleep(40)
+
+    os.system('docker-compose up -d')
+
+
+def update(arguments):
+    print 'Updating Docker Network...'
+    deploy_genesis = arguments['genesis']
+
+    os.system('docker kill $(docker ps -q -f network=docker-network_vpcbr)')
+
+    if deploy_genesis:
+        print 'Deploying Genesis...'
+        os.system('docker exec -it builder bash -c '
+                  '"./home/ubuntu/go/protocol/DOCKER-network/default/scripts/genesis.sh"')
+
+    os.system('docker exec -it builder bash -c "cd /home/ubuntu/go/protocol && make install_c"')
+    os.system('docker-compose up -d')
+
+
+def stop(arguments):
+    print 'Stopping Docker Network...'
+    # stop all nodes
+    os.system('docker kill $(docker ps -q -f network=docker-network_vpcbr)')
+    # remove node data if clean flag is set
+    if arguments['clean']:
+        print 'Cleaning Node Data...'
+        os.system('docker exec -it builder bash -c "rm -rf /opt/data/devnet/*-Node"')
+    # remove the remaining containers
+    os.system('docker container kill $(docker ps -q)')
+    os.system('docker system prune -f')
+
+
+def run(arguments):
+    container = arguments['container']
+    cmd = arguments['cmd']
+    docker_cmd = 'docker exec -it ' + container + ' bash -c \"cd /opt/data/devnet && ' + cmd + '\"'
```

Comment:
Is it a good idea to use 'bash' as default CMD?

Reply:
I think it's ok, since the image is built on top of ubuntu

Reply:
Yea, the "docker exec" command will execute inside the ubuntu image

Reply:
If I just type "python dockernet run 0-Node", the command prompted error and requires a command. It worked if I typed "python dockernet run 0-Node bash".

Reply:
that is correct behaviour, the "run" command is meant to execute a bash command inside the given container